load-agent/locust-files/locustMediatorIssue.py

```diff
@@ -1,41 +1,22 @@
-import os
+import time
 
-from constants import standard_wait
-from locust import SequentialTaskSet, User, task
-from locustClient import CustomClient
+from constants import deviation_wait, standard_wait
+from locust import task
+from locustConnection import ConnectionUserBehaviour
+from locustCustom import CustomLocust
 
-WITH_MEDIATION = os.getenv("WITH_MEDIATION")
 
-class CustomLocust(User):
-    abstract = True
+class UserBehaviour(ConnectionUserBehaviour):
     def __init__(self, *args, **kwargs):
-        super().__init__(*args,**kwargs)
-        self.client = CustomClient(self.host)
-
-class UserBehaviour(SequentialTaskSet):
-    def on_start(self):
-        self.client.startup(withMediation=bool(WITH_MEDIATION))
-
-    def on_stop(self):
-        self.client.shutdown()
-
-    @task
-    def get_invite(self):
-        invite = self.client.issuer_getinvite()
-        self.invite = invite
-
-    @task
-    def accept_invite(self):
-        self.client.ensure_is_running()
-        connection = self.client.accept_invite(self.invite['invitation_url'])
-        if connection is not None:
-            self.connection = connection
+        super().__init__(*args, **kwargs)
 
     @task
     def receive_credential(self):
         self.client.ensure_is_running()
+        for invite in self.invites:
+            self.client.receive_credential(invite["connection_id"])
+            time.sleep(deviation_wait(0.1, 0.3))
```

Comment:
Should this be hardcoded or configurable? What is this wait for?

Reply:
I don't really want to add a configuration for this. It's just a short random time so the issuance for each connection doesn't happen basically at the exact same time. I had thought about using the same configured time between locust tasks (0.1s-1s default) as the wait here, but it seemed a bit slow. 

I think I'll change it back to that though instead of these hardcoded values. 0.1 to 1 second is still pretty quick for a single agent issuing. And then the `LOCUST_MIN_WAIT` and `LOCUST_MAX_WAIT` env variables could be used to speed it up.